booking-up-for-beauty/smunilla/booking_up_for_beauty.go

```diff
@@ -0,0 +1,57 @@
+package booking
+
+import (
+	"fmt"
+	"time"
+)
+
+func parseDate(date string) time.Time {
+	var layouts = []string{
+		"1/2/2006 15:04:05",
+		"January 2, 2006 15:04:05",
+		"Monday, January 2, 2006 15:04:05",
+	}
+
+	for _, layout := range layouts {
+		t, err := time.Parse(layout, date)
+		if err == nil {
+			return t
+		}
+	}
+
+	panic(fmt.Sprintf("Problem parsing date: %s", date))
+}
+
+// Schedule returns a time.Time from a string containing a date
+func Schedule(date string) time.Time {
+	return parseDate(date)
+}
+
+// HasPassed returns whether a date has passed
+func HasPassed(date string) bool {
+	return time.Now().After(parseDate(date))
+}
+
+// IsAfternoonAppointment returns whether a time is in the afternoon
+func IsAfternoonAppointment(date string) bool {
+	apptHour := parseDate(date).Hour()
+
+	return apptHour >= 12 && apptHour <= 18
+}
+
+// Description returns a formatted string of the appointment time
+func Description(date string) string {
+	appt := parseDate(date)
+	return fmt.Sprintf("You have an appointment on %s, %s %d, %d, at %d:%d.",
```

Comment:
You can use `appt.Format("You have an appointment on Monday, January 2, 2006, at 15:04.")` to maximize Go convenience things. It feels weird since you will see a string with a real date, but it works xD

Reply:
I wasn't sure how I felt about that versus this way for readability. But yeah, they both work.